### ISSUE TITLE
Shape from two stops - straight line

### ISSUE THREAD

**Issue by mgmembiela** [ad-freiburg/pfaedle#38]
Hi we need claculate chape from two points (stops) and the results is a sstraight line, with three stops ot more the result is a good shape. Any known problems to make a shape with only **two points**? Could it be too far away?
Thanks!
![L0110](https://user-images.githubusercontent.com/118879120/203487128-c14dc90a-06ef-47de-8a30-d7d95c9e6a58.PNG)

**Comment by patrickbr** [comment]
Hi, this should be no problem - are you sure that there is actually a possible route through the OSM data connecting the two stops? A straight line is a fallback if no matching route was found. Could you give me the coordinates, the route type, and the arrival and departure times at the first and last station?

**Comment by mgmembiela** [comment]
Hi, yes there is an OSM route (attached image):

**coors**
41.379550934,2.1383321285
41.594032288,1.8397530317
**route_type** = 3 (BUS)

**stops**
stop_id,stop_code,stop_name,stop_desc,stop_lat,stop_lon,zone_id,stop_url,location_type,parent_station,stop_timezone,wheelchair_boarding,platform_code
PF08019019,,Barcelona (Sants),,41.379573822,2.1383149624,,,0,,,0,
PF08127005,,Monestir de Montserrat,,41.594032288,1.8397530317,,,0,,,0,

**stop_times**
trip_id,arrival_time,departure_time,stop_id,stop_sequence,stop_headsign,pickup_type,drop_off_type,shape_dist_traveled,timepoint
114157145,17:00:30,17:00:30,PF08127005,0,,0,0,0,1
114157145,18:15:30,18:15:30,PF08019019,1,,0,3,34496.375,1
![osm](https://user-images.githubusercontent.com/118879120/203518029-95b1d26f-0437-41ce-af09-38fddb997abd.PNG)

**Comment by mgmembiela** [comment]
HI, it is working for you? thanks.

**Comment by patrickbr** [comment]
Yes, sorry for the late reply - I am currently debugging this, your data was very helpful :)

**Comment by mgmembiela** [comment]
> Yes, sorry for the late reply - I am currently debugging this, your data was very helpful :)

Hi! any idea?

**Comment by zabuTNT** [comment]
I had a similar issue. 
Probably you have to customize the default pfaedle.conf.

Check this issue, too:
https://github.com/ad-freiburg/pfaedle/issues/29
Maybe there is an issue on OSM about the oneway directions, or you have to change penalties.

![2024-04-23-135416_1093x506_scrot](https://github.com/ad-freiburg/pfaedle/assets/5268372/c52d1eeb-b0cb-41d5-a9cd-821dcd7d3dec)

**Comment by doncicuto** [comment]
Hi there, pfaedle is awesome thanks a lot for it. I too have found problems with trips that only have two stops on it, I get a straight line (either for rails or for buses). If the same route has more stops the shape is created fine.

I've tried to debug this issue without luck. About the mentioned penalties would you give me a hint on what I could try to change?

Thanks in advance
